addons/sale/models/payment.py

```diff
@@ -110,6 +110,8 @@ def _reconcile_after_transaction_done(self):
             tx._check_amount_and_confirm_order()
         # send order confirmation mail
         sales_orders._send_order_confirmation_mail()
+        # Create invoices with the correct timezone by preventing it from defaulting to UTC when the user is not logged in.
+        self = self.with_context(tz=self.env.context.get('tz') or self.partner_id.tz or self.sale_order_ids.user_id.tz)
```

Comment:
Well, in fact, this is wrong now that i read it more attentively.
1) self can contain multiple records, which means `self.partner_id.tz` will raise.  And that the context change here is not right, cause you may want different tz contexts for different transactions reconciliations.
2) is the fallback on the salesman really useful?  Seems strange to me, and would break if multiple orders are linked to the transaction. why can't you follow the same tz logic as in payment and fallback on the company tz ?

Reply:
@Feyensv you're right I haven't thought about the case of multiple transactions.
I was using the salesman before thinking of using the company tz, it seems redundant now. I'll delete the changes made to the sale module and its tests and edit the fix on payment to apply the tz for each transaction individually.